test with vcr

dpprdan opened the pull request:
## Description

This will stub HTTP requests with {vcr}, i.e. tests will not run against the OpenCage API directly, but against recorded responses from the API. 

TODOs

- [x] mask test keys with vcr
- [x] vcr'ify tests of deprecated functions
- [x] check http testing book for best practices (check cassette names for example) https://books.ropensci.org/http-testing/vcr.html#vcr
- [x] check whether existing tests with skip_if_offline() or skip_if_no_key() can be vcr'ified
- [x] split up long cassettes, e.g. *_tidyeval.yml?
- [x] lintr?
- [x] mention vcr in contributing.md

closes #48

dpprdan committed:
start with vcr

dpprdan committed:
vcr oc_get

dpprdan committed:
vcrify all the things

dpprdan committed:
mind your lint

dpprdan committed:
skip_if_no_key and match on uri only

dpprdan committed:
memoise::forget

dpprdan committed:
vcr-mask OpenCage test keys as well so we can identify them in cassettes

dpprdan committed:
lint

dpprdan committed:
add vcr's .gitattributes

dpprdan committed:
rename helper- to setup-opencage.R, use vcr_test_path() and check_cassette_names()

dpprdan committed:
use vcr with tests of deprecated functions, delete duplicate tests in test-opencage_forward.R

dpprdan committed:
re-record vcr_cassettes

dpprdan committed:
check if a key is present before sending a query

dpprdan committed:
rename helper-opencage.R to setup-opencage.R

dpprdan committed:
add NEWS item

dpprdan committed:
set fake API key and configure rate limit for vcr

dpprdan committed:
test against default rate limit

dpprdan committed:
drop skip_if_no_key() and skip_of_oc_offline()

dpprdan committed:
vcrify oc_get() tests

dpprdan committed:
Merge branch 'master' into feat_vcr

dpprdan committed:
reduce rate limit with vcr also when key is available

dpprdan committed:
split up large cassettes and other style improvements

dpprdan committed:
mention vcr in contributing.md

dpprdan committed:
moar cassettes

dpprdan committed:
bump required memoise version because https://github.com/r-lib/memoise/pull/61

dpprdan committed:
Merge branch 'master' into feat_vcr

dpprdan commented:
I think I am done here. 😅 
@maelle, @jessesadler if one or both of you have got time to review, it would be very much appreciated! 😃

maelle commented:
I could have a look but after March the 8th. Awesome work in any case! (I just had a quick look at the setup, the keys seem well hidden :-) )

dpprdan commented:
> the keys seem well hidden :-) 

Yeah, I hid the test keys as well so they are identifiable (as in "which key is used here?") in the cassettes and vcr's error messages.

dpprdan commented:
@maelle There are a few empty cassettes because the same API request was made earlier (with different functions/arguments) and cached then. I am [not sure](https://github.com/ropensci/vcr/issues/224#issuecomment-791650272) whether this is the best approach.

maelle commented on tests/testthat/setup-opencage.R:
```diff
@@ -1,20 +1,47 @@
+    dir = vcr_dir,
+    match_requests_on = "uri", # method not necessary, we only make GET requests
+    filter_sensitive_data =
+      list(
```
This could be more specific I think https://github.com/ropensci/vcr/commit/eeb5c9e9331616101fd1a673095a2d087d815af6

dpprdan replied:
Would [that someone](https://github.com/ropensci/vcr/issues/212#issuecomment-792607578) be me, because I am not sure I can follow. 😄 
How is `filter_query_parameters` better here? From what I understand, it is not possible to distinguish between different keys with `filter_query_parameters`, is it?

maelle replied:
lol yes it was you and I feel bad not having provided more details at the time :joy: https://docs.ropensci.org/vcr/articles/configuration.html#filter-query-parameters

I am realizing it can also be less specific so maybe not that useful here.

dpprdan committed:
bump suggested vcr version to current dev

dpprdan committed:
re-use cassettes and delete empty ones

dpprdan committed:
clear cache before running tests

dpprdan commented:
@maelle empty cassettes are no longer an issue. This needs the current dev version of vcr though, so we can't push this to CRAN before vcr v0.7 is released.

maelle commented:
sounds good!

dpprdan committed:
Merge branch 'main' into feat_vcr

dpprdan committed:
re-record cassette for min_confidence test

dpprdan committed:
configure OS and RSPM only once in matrix, include OS version in cache key

dpprdan committed:
OPENCAGE_KEY not necessary in R-CMD-check workflow anymore

all API requests run against vcr cassettes; a fake key is provided in setup_opencage.R

dpprdan committed:
add weekly test with real API requests

dpprdan commented:
closing this in favour of #137
